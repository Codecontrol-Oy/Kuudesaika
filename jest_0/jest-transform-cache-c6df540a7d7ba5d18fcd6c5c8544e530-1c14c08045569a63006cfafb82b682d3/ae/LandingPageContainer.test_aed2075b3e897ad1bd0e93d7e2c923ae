'use strict';



jest.mock('react-router', function () {return { browserHistory: { push: jest.fn() } };});var _react = require('react');var _react2 = _interopRequireDefault(_react);var _LandingPageContainer = require('./LandingPageContainer');var _LandingPageContainer2 = _interopRequireDefault(_LandingPageContainer);var _sinon = require('sinon');var _sinon2 = _interopRequireDefault(_sinon);var _reactRouter = require('react-router');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe("<LandingPageContainer />", function () {

    it("Matches snapshot", function () {
        var landingPageContainer = shallow(
        _react2.default.createElement(_LandingPageContainer2.default, null));

        expect(landingPageContainer).toMatchSnapshot();
    });

    it("Should render city cards", function () {
        var landingPageContainer = shallow(
        _react2.default.createElement(_LandingPageContainer2.default, null));

        expect(landingPageContainer.find('Card').length).toEqual(6);
    });

    it("Should redirect to city on click", function () {
        var landingPageContainer = shallow(
        _react2.default.createElement(_LandingPageContainer2.default, null));

        var callback = _sinon2.default.spy(landingPageContainer.instance(), 'onSelectCity');
        var organizationLink = landingPageContainer.find('Card');
        organizationLink.first().simulate('click');
        expect(callback.calledOnce).toEqual(true);
        expect(_reactRouter.browserHistory.push).toHaveBeenCalled();
    });
});