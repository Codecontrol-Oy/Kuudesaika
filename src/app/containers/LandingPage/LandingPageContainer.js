import React from 'react';
import { connect } from 'react-redux';
import Section from 'grommet/components/Section';
import Anchor from 'grommet/components/Anchor';
import Tiles from 'grommet/components/Tiles';
import Card from 'grommet/components/Card';
import Image from 'grommet/components/Image';
import {setCity} from 'Actions';
import {browserHistory} from 'react-router';
import theme from './LandingPageContainer.scss';

export default class LandingPageContainer extends React.Component {
  constructor (props) {
    super(props);
  }

  onSelectCity = (value) => {
    setCity(value);
    browserHistory.push(`/${value}`);
  }

  render () {
    return (
        <Section>
                <Tiles>
                <Card
                    heading={"Helsinki"}
                    thumbnail={<Image alt={"temporary image"} src={'http://lorempixel.com/640/480/city?' + 1} />}
                    headingStrong
                    onClick={() => this.onSelectCity('Helsinki')} 
                    link={<Anchor onClick={() => this.onSelectCity('Helsinki')} label={"siirry"} />}
                    className={theme.cardShadow}/>
                <Card
                    heading={"Espoo"}
                    thumbnail={<Image alt={"temporary image"} src={'http://lorempixel.com/640/480/city?' + 2} />}
                    headingStrong
                    onClick={() => this.onSelectCity('Espoo')}
                    link={<Anchor onClick={() => this.onSelectCity('Espoo')} label={"siirry"} />}
                    className={theme.cardShadow}/>
                <Card
                    heading={"Turku"}
                    thumbnail={<Image alt={"temporary image"} src={'http://lorempixel.com/640/480/city?' + 3} />}
                    headingStrong
                    onClick={() => this.onSelectCity('Turku')}                  
                    link={<Anchor onClick={() => this.onSelectCity('Turku')} label={"siirry"} />}
                    className={theme.cardShadow}/>
                <Card
                    heading={"Oulu"}
                    thumbnail={<Image alt={"temporary image"} src={'http://lorempixel.com/640/480/city?' + 4} />}
                    headingStrong
                    onClick={() => this.onSelectCity('Oulu')}
                    link={<Anchor onClick={() => this.onSelectCity('Oulu')} label={"siirry"} />}
                    className={theme.cardShadow}/>
                <Card
                    heading={"Tampere"}
                    thumbnail={<Image alt={"temporary image"} src={'http://lorempixel.com/640/480/city?' + 5} />}
                    headingStrong
                    onClick={() => this.onSelectCity('Tampere')}                    
                    link={<Anchor onClick={() => this.onSelectCity('Tampere')} label={"siirry"} />}
                    className={theme.cardShadow}/>
                <Card
                    heading={"Vantaa"}
                    thumbnail={<Image alt={"temporary image"} src={'http://lorempixel.com/640/480/city?' + 6} />}
                    headingStrong
                    onClick={() => this.onSelectCity('Vantaa')}                
                    link={<Anchor onClick={() => this.onSelectCity('Vantaa')} label={"siirry"} />}
                    className={theme.cardShadow}/>
                </Tiles>
        </Section>
    );
  }
}