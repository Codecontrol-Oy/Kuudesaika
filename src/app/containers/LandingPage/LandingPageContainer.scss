.cardShadow {
    box-shadow: 0px 0px 17px -1px rgba(0,0,0,0.5);
    margin: 12px;
}