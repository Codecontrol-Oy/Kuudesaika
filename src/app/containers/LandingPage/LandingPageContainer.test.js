import React from 'react';
import LandingPageContainer from './LandingPageContainer';
import sinon from 'sinon';
import { browserHistory } from 'react-router';
jest.mock('react-router', ()=>({browserHistory: {push: jest.fn()}}))

describe("<LandingPageContainer />", () => {

    it("Matches snapshot", () => {
        const landingPageContainer = shallow(
            <LandingPageContainer />
        );
        expect(landingPageContainer).toMatchSnapshot();
    });

    it("Should render city cards", () => { 
        const landingPageContainer = shallow(
            <LandingPageContainer />
        );
        expect(landingPageContainer.find('Card').length).toEqual(6);
    });

    it("Should redirect to city on click", () => { 
        const landingPageContainer = shallow(
            <LandingPageContainer />
        );
        const callback = sinon.spy(landingPageContainer.instance(),'onSelectCity');
        const organizationLink = landingPageContainer.find('Card');
        organizationLink.first().simulate('click');
        expect(callback.calledOnce).toEqual(true);
        expect(browserHistory.push).toHaveBeenCalled(); 
    });
}) 
