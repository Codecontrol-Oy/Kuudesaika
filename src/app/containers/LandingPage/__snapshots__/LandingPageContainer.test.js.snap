// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPageContainer /> Matches snapshot 1`] = `
<Section
  pad={
    Object {
      "vertical": "medium",
    }
  }
>
  <Tiles
    flush={true}
    justify="start"
    pad="small"
  >
    <Card
      a11yTitle="Card"
      contentPad="medium"
      heading="Helsinki"
      headingStrong={true}
      link={
        <Anchor
          animateIcon={true}
          label="siirry"
          method="push"
          onClick={[Function]}
          tag="a"
        />
      }
      onClick={[Function]}
      textSize="small"
      thumbnail={
        <Image
          alt="temporary image"
          size="medium"
          src="http://lorempixel.com/640/480/city?1"
        />
      }
    />
    <Card
      a11yTitle="Card"
      contentPad="medium"
      heading="Espoo"
      headingStrong={true}
      link={
        <Anchor
          animateIcon={true}
          label="siirry"
          method="push"
          onClick={[Function]}
          tag="a"
        />
      }
      onClick={[Function]}
      textSize="small"
      thumbnail={
        <Image
          alt="temporary image"
          size="medium"
          src="http://lorempixel.com/640/480/city?2"
        />
      }
    />
    <Card
      a11yTitle="Card"
      contentPad="medium"
      heading="Turku"
      headingStrong={true}
      link={
        <Anchor
          animateIcon={true}
          label="siirry"
          method="push"
          onClick={[Function]}
          tag="a"
        />
      }
      onClick={[Function]}
      textSize="small"
      thumbnail={
        <Image
          alt="temporary image"
          size="medium"
          src="http://lorempixel.com/640/480/city?3"
        />
      }
    />
    <Card
      a11yTitle="Card"
      contentPad="medium"
      heading="Oulu"
      headingStrong={true}
      link={
        <Anchor
          animateIcon={true}
          label="siirry"
          method="push"
          onClick={[Function]}
          tag="a"
        />
      }
      onClick={[Function]}
      textSize="small"
      thumbnail={
        <Image
          alt="temporary image"
          size="medium"
          src="http://lorempixel.com/640/480/city?4"
        />
      }
    />
    <Card
      a11yTitle="Card"
      contentPad="medium"
      heading="Tampere"
      headingStrong={true}
      link={
        <Anchor
          animateIcon={true}
          label="siirry"
          method="push"
          onClick={[Function]}
          tag="a"
        />
      }
      onClick={[Function]}
      textSize="small"
      thumbnail={
        <Image
          alt="temporary image"
          size="medium"
          src="http://lorempixel.com/640/480/city?5"
        />
      }
    />
    <Card
      a11yTitle="Card"
      contentPad="medium"
      heading="Vantaa"
      headingStrong={true}
      link={
        <Anchor
          animateIcon={true}
          label="siirry"
          method="push"
          onClick={[Function]}
          tag="a"
        />
      }
      onClick={[Function]}
      textSize="small"
      thumbnail={
        <Image
          alt="temporary image"
          size="medium"
          src="http://lorempixel.com/640/480/city?6"
        />
      }
    />
  </Tiles>
</Section>
`;
